core/src/com/unciv/logic/github/Github.kt

```diff
@@ -99,30 +112,53 @@ object Github {
             zipUrl = repo.direct_zip_url
             tempName = "temp-" + repo.toString().hashCode().toString(16)
         }
-        
+
         var contentLength = 0
-        val inputStream = download(zipUrl) {
-            // We DO NOT want to accept "Transfer-Encoding: chunked" here, as we need to know the size for progress tracking
-            // So this attempts to limit the encoding to gzip only
-            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding
-            // HOWEVER it doesn't seem to work - the server still sends chunked data sometimes 
-            // which means we don't actually know the total length :(
-            it.setRequestProperty("Accept-Encoding", "gzip")
-            
-            val disposition = it.getHeaderField(contentDispositionHeader)
-            if (disposition.startsWith(attachmentDispositionPrefix))
-                modNameFromFileName = disposition.removePrefix(attachmentDispositionPrefix)
-                    .removeSuffix(".zip").replace('.', ' ')
-            // We could check Content-Type=[application/x-zip-compressed] here, but the ZipFile will catch that anyway. Would save some bandwidth, however.
-            
-            contentLength = it.getHeaderField("Content-Length")?.toInt()
-                ?: 0 // repo.length is a total lie
-        } ?: return null
+        var redirectCount = 0
+        var currentUrl = zipUrl
+        var inputStream: InputStream? = null
+        while (redirectCount < 2) {
```

Comment:
This is good, but this function is getting too big - extract to a different function called getStreamFollowingRedirects or something